File: Package/ios/RNLocalNotifierModule.m
//
//  RNLocalNotifierModule.m
//  RNLocalNotifierModule
//
//  Copyright © 2022 Ahmed Essam. All rights reserved.
//

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNLocalNotifierModule, NSObject)
RCT_EXTERN_METHOD(requestAuthorization: (RCTPromiseResolveBlock)resolve rejecter: (RCTPromiseRejectBlock)reject)
RCT_EXTERN_METHOD(sendNotification: (NSDictionary *)notifierObject)
RCT_EXTERN_METHOD(scheduleNotification: (NSDictionary *)notifierObject)

@end

File: Package/ios/RNLocalNotifierModule.swift
import Foundation

@objc(RNLocalNotifierModule)
class RNLocalNotifierModule: NSObject {

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }

   @objc
  func requestAuthorization(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock ) -> Void {
    
              NotificationCenter.shared.authorize { granted in
                resolve(granted)
              }
  
  }

  @objc
  func sendNotification(_ notifierObject: NSDictionary) -> Void {

    NotificationCenter.shared.send(notifierObject: notifierObject)
  }

   @objc
  func scheduleNotification(_ notifierObject: NSDictionary) -> Void {

    NotificationCenter.shared.schedule(notifierObject: notifierObject)
  }

}
